main.cc

```diff
@@ -0,0 +1,86 @@
+#include <array>
+#include <cmath>
+#include <iostream>
+#include <vector>
+
+int main() {
+
+	// Initialize the variables
+	double vertical_spacing, horizontal_spacing;
+	int num_vertical_points, num_horizontal_points;
```

Comment:
number of points are never going to be negative so use `unsigned`. Also declare it where you use it.